mindsdb/integrations/handlers/hubspot_handler/hubspot_handler.py

```diff
@@ -49,48 +50,501 @@ def __init__(self, name: str, **kwargs):
+        """Get description for a table."""
+        descriptions = {
+            "companies": "HubSpot companies data including name, industry, location and other company properties",
+            "contacts": "HubSpot contacts data including email, name, phone and other contact properties",
+            "deals": "HubSpot deals data including deal name, amount, stage and other deal properties",
+        }
+        return descriptions.get(table_name, f"HubSpot {table_name} data")
+
+    def _estimate_table_rows(self, table_name: str) -> Optional[int]:
+        """Estimate number of rows in a table using HubSpot API."""
+        try:
+            if table_name == "companies":
+                companies = self.connection.crm.companies.get_all(limit=100)
+                return len(list(companies)) if companies else 0
+            elif table_name == "contacts":
+                contacts = self.connection.crm.contacts.get_all(limit=100)
+                return len(list(contacts)) if contacts else 0
+            elif table_name == "deals":
+                deals = self.connection.crm.deals.get_all(limit=100)
+                return len(list(deals)) if deals else 0
+        except Exception as e:
+            logger.warning(f"Could not estimate rows for {table_name}: {str(e)}")
+        return None
+
+    def _get_columns_with_statistics(self, table_name: str) -> List[Dict[str, Any]]:
+        """Get detailed column information with comprehensive statistics."""
+        try:
+            sample_data = None
+
+            if table_name == "companies":
+                sample_data = list(self.connection.crm.companies.get_all(limit=1000))
+            elif table_name == "contacts":
+                sample_data = list(self.connection.crm.contacts.get_all(limit=1000))
+            elif table_name == "deals":
+                sample_data = list(self.connection.crm.deals.get_all(limit=1000))
+
+            columns_info = []
+
+            if sample_data and len(sample_data) > 0:
+                sample_size = len(sample_data)
+                logger.info(f"Analyzing {sample_size} records for {table_name} column statistics")
+
+                all_properties = set()
+                for item in sample_data:
+                    if hasattr(item, "properties") and item.properties:
+                        all_properties.update(item.properties.keys())
+
+                id_stats = self._calculate_column_statistics("id", [item.id for item in sample_data])
+                columns_info.append(
+                    {
+                        "COLUMN_NAME": "id",
+                        "DATA_TYPE": "VARCHAR",
+                        "IS_NULLABLE": False,
+                        "COLUMN_DEFAULT": None,
+                        "COLUMN_DESCRIPTION": "Unique identifier for the record (Primary Key)",
+                        "IS_PRIMARY_KEY": True,
+                        "IS_FOREIGN_KEY": False,
+                        "NULL_COUNT": id_stats["null_count"],
```

Comment:
Check methods meta_get_columns and meta_get_column_statistics from [here](https://github.com/mindsdb/mindsdb/blob/main/mindsdb/integrations/libs/api_handler.py#L348) to get required fields list

Reply:
done